### PR TITLE
feat: add monaco editor for InfluxQL

### PR DESCRIPTION
**Pull request by appletreeisyellow**
Closes https://github.com/influxdata/idpe/issues/17070

This PR creates InfluxQL editor component, including hotkeys, syntax, and theme

This work is behind the feature flag `influxqlUI`

Note that the test for syntax and tokenizer is not done yet. It is tracked in https://github.com/influxdata/idpe/issues/17173

## Before
Screen Shot 2023-02-23 at 2 26 08 PM

## After
Screen Shot 2023-02-23 at 2 27 10 PM

### Checklist

Authors and Reviewer(s), please verify the following:

- [x] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [x] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [x] Documentation updated or issue created (provide link to issue/PR)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] Feature flagged, if applicable

### CODE REVIEW

File: src/languageSupport/languages/influxql/influxql.ts [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+      [/[$][+-]*\d*(\.\d*)?/, 'literal.number'],
+      [/((\d+(\.\d*)?)|(\.\d+))([eE][\-+]?\d+)?/, 'literal.number'],
+    ],
+    strings: [[/'/, {token: 'literal.string', next: '@string'}]],
+    string: [
+      [/[^']+/, 'literal.string'],
+      [/''/, 'literal.string'],
+      // support single quotes with backslash escape
+      // https://docs.influxdata.com/influxdb/v2.6/reference/syntax/influxql/spec/#strings
+      [/^'((?:\\.|[^\\'])*)'$/, 'literal.string'],
```
Comment (wiedld) [inline src/languageSupport/languages/influxql/influxql.ts:282]: This is not working properly:
`'str\'ing'`

File: src/languageSupport/languages/influxql/influxql.ts [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+      [/'/, {token: 'literal.string', next: '@pop'}],
+    ],
+    complexIdentifiers: [
+      [/"/, {token: 'identifier.quote', next: '@quotedIdentifier'}],
+    ],
+    quotedIdentifier: [
```
Comment (wiedld) [inline src/languageSupport/languages/influxql/influxql.ts:288]: Quoted identifiers can have an internal `"`:
`"air\"Sensors"`

https://docs.influxdata.com/influxdb/v2.6/reference/syntax/influxql/spec/#identifiers

**Review by wiedld — CHANGES_REQUESTED** [review]
Amazing, awesome job! 🎉 

Have a few small changes in the grammar rules.

File: src/languageSupport/languages/influxql/influxql.ts [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    ],
+    comments: [
+      [/--+.*/, 'comment'],
+      [/\/\*/, {token: 'comment.quote', next: '@comment'}],
+    ],
+    comment: [
+      [/[^*/]+/, 'comment'],
+      // Not supporting nested comments, as nested comments seem to not be standard?
+      // i.e. http://stackoverflow.com/questions/728172/are-there-multiline-comment-delimiters-in-sql-that-are-vendor-agnostic
+      // [/\/\*/, { token: 'comment.quote', next: '@push' }],    // nested comment not allowed :-(
```
[the remark this answers is not in the document]
Comment (appletreeisyellow) [reply src/languageSupport/languages/influxql/influxql.ts:266]: Nice catch! Done

File: src/languageSupport/languages/influxql/influxql.ts (hunk 1 above)
Comment (appletreeisyellow) [reply src/languageSupport/languages/influxql/influxql.ts]: Done

File: src/languageSupport/languages/influxql/influxql.ts (hunk 2 above)
Comment (appletreeisyellow) [reply src/languageSupport/languages/influxql/influxql.ts:293]: Done

**Review by wiedld — APPROVED** [review]

**Pull request merged**